### ISSUE TITLE
(about nextSelector) how to define it in pages where there's no 'Next' page, but only links to page 1,2,etc

### ISSUE THREAD

**Issue by darkred** [pklauzinski/jscroll#75]
I've been trying to make this userscript in order to use jScroll it https://openuserjs.org/ 

```
$('table')[0].jscroll({
  nextSelector: $('ul.pagination > li > a')[1],
  contentSelector: $('tbody:nth-of-type(1) > tr.tr-link').slice(0,24),
  callback: function () {
     $('ul.pagination')[1].remove();
  }
});
```

First of all, in that page, the "Scripts" table  is the 1st (the other is the "Announcements" table),
so I used `$('table')[0].jscroll({`. 
But, unfortunately I get `$(...)[0].jscroll is not a function` in FIrefox's Browser Console, 
therefore the script cannot be executed.
**Shouldn't this work, though?**

Also, the main problem is that, in that page, there's no 'Next' page, but only links to p.1,2,etc.
So, I used `nextSelector: $('ul.pagination > li > a')[1],` as a reference, 
but, **how could I select each `nextSelector`?** 
(i.e. if I'm on page 1, the nextSelector will be the `a` to page 2. If  then it paginates 2, it will be 3, etc)

<br>
_The other selectors are:
  `contentSelector: $('tbody:nth-of-type(1) > tr.tr-link').slice(0,24),`
i.e. the the 24 rows of the 1st table, 
and the `callback` function removes the 2nd "group" of links to next pages(i.e the one in the bottom), 
that is `$('ul.pagination')[1]`._

**Comment by pklauzinski** [comment]
The reason your call to `jscroll` isn't working is because `$(...)[0]` is not a jQuery object, it is just the DOM node itself. Instead, you should do something like `$(...).first().jscroll` if you want to set it on only the first element in the array.

**Comment by pklauzinski** [comment]
As for your `nextSelector`, jScroll only looks for the first instance of it, as there should only be one link to the next page, as shown in this line directly from the source:

```js
_$next = $e.find(_options.nextSelector).first()
```

Also, make sure that your `nextSelector` is just the selector *text*, and not a jQuery object itself.

**Comment by pklauzinski** [comment]
In looking at the page, you could potentially find the next selector by using the CSS *adjacent sibling* selector. Try this:

```js
nextSelect: 'ul.pagination > li.active + li > a'
```

This will access the next page link after the currently *active* page.

**Comment by darkred** [comment]
Thanks for replying.

So, I've tried this:

```
$('table').first().jscroll({
  nextSelector: 'ul.pagination > li.active + li > a',
  contentSelector: 'col-sm-8 .panel .table tr.tr-link, ul.pagination',
});
```
but unfortunately it doesn't work (it doesn't paginate not even page 2).

**Comment by pklauzinski** [comment]
It looks like the pagination is not actually within the table element. jScroll looks for the `nextSelector` *within* the jScroll selector, so you probably want to actually call jScroll on the parent of both elements. It looks like `.col-sm-8` would be your best bet.

**Comment by pklauzinski** [comment]
Try this:

```js
$('.col-sm-8').first().jscroll({
  nextSelector: 'ul.pagination > li.active + li > a',
  contentSelector: '.panel .table tr.tr-link, ul.pagination'
});
```

**Comment by darkred** [comment]
With your latest code, the pages are indeed paginated.

In order to remove the nav buttons I added this to your code: 
```
callback: function () {
        $('ul.pagination').remove();
    }
```
I'd prefer the pages be paginated to table as one continuous table.
The only problem now is that the table "Announcements" (on the right), is also included in every pagination (of page 3 and on).
Could this be avoided?

**Comment by pklauzinski** [comment]
Instead of removing the nav buttons with your callback function, you can instead use the `pagingSelector` option in jScroll, which searches the `nextSeletor` parent chain for that selector and will hide the pagination controls if found:

```js
$('.col-sm-8').first().jscroll({
    nextSelector: 'ul.pagination > li.active + li > a',
    contentSelector: '.panel .table tr.tr-link, ul.pagination',
    pagingSelector: 'ul.pagination'
});
```

You could alternatively use the `callback` option to remove the "Announcements" table with each request.

**Comment by darkred** [comment]
`pagingSelector ` unfortunately didn't hide the nav buttons.
But, if I add this:

```
callback: function () {
        $('ul.pagination').remove();
    }
```
then they are removed.
But still the rows of each next paginated page are not merged with the table from the previous page.
Is it possible this to be fixed?

<br>
> You could alternatively use the `callback` option to remove the "Announcements" table with each request.

Currently the 5 rows from the "Announcements" table (i.e. except the header) are added in each paginated page (from page 3 and on).
What selector can I use for these 5 lines to be removed via the `callback` option?
They don't have any `id` or `class` name.

FWIW their CSS selectors are like this:  
`div.jscroll-added:nth-child(5) > tr:nth-child(26)` up to
`div.jscroll-added:nth-child(5) > tr:nth-child(30)`. Then from
`div.jscroll-added:nth-child(6) > tr:nth-child(26)` up to 30 ,etc

**Comment by hakt0-r** [comment]
Hi there, I though about opening another issue but my question is very similar to this thread, thus the post here. 

I'll begin by saying that I am a newbie in JS/jQuery/everything but I am a fast learner. I have already managed to use your example in a few tampermonkey quickie scripts for a few of sites and on this one I have hit a snag [https://www.newscientist.com/issues/](url)

This is my code...
```
$('.article-copy').jscroll({
//$('.magazine-archive-issues').jscroll({
    debug: true,
    autoTrigger: true,
    padding: 100,
//    nextSelector: 'entry-form-drop-down-menu.val < entry-form-drop-down-menu.options.selected',
    nextSelector: '/issues/+(entry-form-drop-down-menu.val-1)', 
    contentSelector: '.article-copy',
//    contentSelector: '.magazine-archive-issues',
    pagingSelector : 'section.magazine-archive-browser entry-form',
});
```
So, here is where I am going nuts, I think this is like the unordered list example and that I am telling the nextSelector to set itself up to point to the next value down the options list. So far I have tried the following to no success...
-- using the DOM (document.getElementbyClassname(...), epic fail!
-- using entry-form-drop-down-menu.value instead of val
-- using assignments entry-form-drop-down-menu.val(2)
-- using calculated assignments entry-form-drop-down-menu.val(entry-form-drop-down-menu.val - 1)
etc...
spent a few good hours on this, and now I am spent and I turn to your council, its a pickle...

Another twister, the pagingSelector, when does that get rid of the pagination control object, is it on the "page 1" or page 2 onwards? I cant seem to find much info on this. :-)

I hope that you can help mate. Cheers in advance.

**Comment by pklauzinski** [comment]
Yes, list elements do not work with jScroll at this time because of the way it wraps sets of content. This is being addressed in v3.0.

**Comment by darkred** [comment]
Thank you.
